pkg/domain/infra/abi/play.go

```diff
@@ -509,11 +513,29 @@ func (ic *ContainerEngine) PlayKube(ctx context.Context, body io.Reader, options
 		}
 
 		report.ServiceContainerID = serviceContainer.ID()
+	} else if serviceContainer != nil {
+		// No containers started, make sure to stop the service container.
+		// Note because the pods still do exists and are not removed by default we cannot remove it.
+		if err := serviceContainer.StopWithTimeout(0); err != nil {
+			logrus.Errorf("Failed to stop service container: %v", err)
+		}
 	}
 
 	return report, nil
 }
 
+// setRanContainers sets the ranContainers bool to true if at least one container was successfully started.
+func setRanContainers(ranContainers *bool, r *entities.PlayKubeReport) {
```

Comment:
Since this method is only used in `PlayKube` and actually only changes `ranContainers` defined in it, will it be simpler to define it as a closure inside the function so that it can access the field (like we do in the `process` function in `quadlet/main.go`?

Reply:
yes sounds good